### PR TITLE
Skip local GraalVM artifacts when not building from source

### LINKED ISSUE #570: CI: Quarkus build fails when using builder images.
## Description
https://github.com/graalvm/mandrel/issues/526 adds support for building with locally installed graalvm maven artefacts. This seems to break builds using builder images, which don't build mandrel/graal from source. The CI check needs to skip downloading those artefacts when an builder image CI build is being triggered.

See this example failure:
```
Starting download for org-graalvm-artefacts--2.13.8.Final-quarkusio-quarkus-graalvm-mandrel-graal-master-master-graalvm-mandrel-packaging-mandrel-source-17-ga-quay.io-quarkus-ubi-quarkus-mandrel-builder-image-22.3-java17-null-graalvm-mandrel-null-null
Error: Unable to find any artifacts for the associated workflow
```
https://github.com/graalvm/mandrel/actions/runs/6168721678/job/16741701212#step:7:24

### PR DESCRIPTION
**Pull request by zakkak**
Closes: https://github.com/graalvm/mandrel/issues/570

### CODE REVIEW

File: .github/workflows/base.yml [hunk 1]
```diff
@@ -393,21 +393,25 @@ jobs:
         distribution: 'temurin'
         java-version: '11'
     - name: Download GraalVM Maven Repo
+      if: ${{ fromJson(needs.build-vars.outputs.build-from-source) == true && needs.build-vars.outputs.distribution == 'mandrel' && inputs.builder-image == 'null'}}
```
Comment (jerboaa) [inline .github/workflows/base.yml]: `distribution == graalvm` builds those too and we should download them in that case. So it should be "mandrel or graalvm" whatever the syntax for that is.

File: .github/workflows/base.yml [hunk 2]
```diff
@@ -393,21 +393,25 @@ jobs:
...
       uses: actions/download-artifact@v3
       with:
         name: org-graalvm-artefacts-${{ needs.get-test-matrix.outputs.artifacts-suffix }}
         path: .
     - name: Download GraalVM Maven Version
+      if: ${{ fromJson(needs.build-vars.outputs.build-from-source) == true && needs.build-vars.outputs.distribution == 'mandrel' && inputs.builder-image == 'null'}}
```
Comment (jerboaa) [inline .github/workflows/base.yml]: Same, re: `graalvm`

File: .github/workflows/base.yml (hunk 1 above)
Comment (jerboaa) [reply .github/workflows/base.yml]: Perhaps we can just drop the condition on `mandrel`. `builder image == null && build from source == true` should be sufficient?

File: .github/workflows/base.yml [hunk 3]
```diff
@@ -418,6 +422,12 @@ jobs:
         cd -
         find ~/.m2/repository/org/graalvm | sort > maven_graalvm_after_build.txt
         diff -u maven_graalvm_before_build.txt maven_graalvm_after_build.txt
+    - name: Build quarkus with default graalvm version
+      if: ${{ fromJson(needs.build-vars.outputs.build-from-source) == false || needs.build-vars.outputs.distribution != 'mandrel' || inputs.builder-image != 'null'}}
```
Comment (jerboaa) [inline .github/workflows/base.yml]: Need to account for graalvm too.

**Review by jerboaa — COMMENTED** [review]
Do we have a CI run testing this? At least the local build one for `mandrel` & `graalvm`.

File: .github/workflows/base.yml (hunk 1 above)
Comment (zakkak) [reply .github/workflows/base.yml]: Yes

**Comment by zakkak** [comment]
> Do we have a CI run testing this? At least the local build one for `mandrel` & `graalvm`.

mandrel from source:  https://github.com/zakkak/mandrel/actions/runs/6185883989
graalvm from source: https://github.com/zakkak/mandrel/actions/runs/6185892130
quay.io/quarkus/ubi-quarkus-mandrel-builder-image:jdk-17: https://github.com/zakkak/mandrel/actions/runs/6185839831

File: .github/workflows/base-wrapper.yml [hunk 4]
```diff
@@ -65,7 +65,7 @@ on:
 
 jobs:
   delegate:
-    uses: graalvm/mandrel/.github/workflows/base.yml@default
+    uses: zakkak/mandrel/.github/workflows/base.yml@2023-09-14-fix-570
```
Comment (jerboaa) [inline .github/workflows/base-wrapper.yml]: I take it this will be reverted once CI completes?

**Review by jerboaa — APPROVED** [review]
LGTM. Please wait for CI before you merge. Thanks!

File: .github/workflows/base-wrapper.yml (hunk 4 above)
Comment (zakkak) [reply .github/workflows/base-wrapper.yml]: Yes, I better not forget it :)

**Comment by zakkak** [comment]
CI runs are good, TEMP commit dropped. I am merging.

**Pull request merged**